src/ArmTemplates/Common/Templates/Abstractions/Template.cs

```diff
@@ -1,16 +1,28 @@
-﻿using Newtonsoft.Json;
+﻿using Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Extensions;
+using Newtonsoft.Json;
 using System.Collections.Generic;
 
 namespace Microsoft.Azure.Management.ApiManagement.ArmTemplates.Common.Templates.Abstractions
 {
     public class Template
```

Comment:
This looks like a DTO. Are we not breaking anything by renaming things here?

Reply:
I have created the default `CamelCasePropertyNamesContractResolver` to be used elsewhere
```
static readonly JsonSerializerSettings DefaultJsonSerializationSettings = new JsonSerializerSettings
        {
            ContractResolver = new CamelCasePropertyNamesContractResolver(),
            Formatting = Formatting.Indented,
            NullValueHandling = NullValueHandling.Ignore
        };
```

when the refactoring is done, every serialization\deserialiation will be working due to this code.

I decided to go with this approach instead of setting `[JsonProperty("bla-bla-bla")]` for every property. It will be quite painful :D